package/index.ts

```diff
@@ -317,15 +317,25 @@ export class UploadArea {
     }
   }
 
-  start(autoClear: boolean = false) {
-    if (this.options.manualStart && this.fileList) {
-      this.putFilesToQueue();
-      if (autoClear) this.clear();
+  start(autoClear: boolean = false, files?: IUploadFile[]) {
+    if (this.options.manualStart && (files || this.fileList)) {
+      this.putFilesToQueue(files);
+      if (autoClear) this.clear(files);
     }
   }
 
-  clear() {
-    this.fileList = null;
+  clear(files?: IUploadFile[]) {
+    if (this.fileList && files !== undefined) {
+      for (let i = 0; i < files.length; ++i) {
+        for (let j = 0; j < this.fileList.length; ++j) {
+          if (files[i] === this.fileList[j]) {
+            this.fileList.splice(j--, 1);
```

Comment:
Here you change the the list you iterate in. Please, cover this code with test and fix the issue.

Reply:
Changed.

I had added `uploadArea.spec.ts` previously. Are there some more tests I should have added?